library/alloc/src/boxed.rs

```diff
@@ -233,6 +233,27 @@ pub struct Box<
     #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
 >(Unique<T>, A);
 
+/// Constructs a `Box<T>` by calling the `exchange_malloc` lang item and moving the argument into
+/// the newly allocated memory. This is an intrinsic to avoid unnecessary copies.
+///
+/// This is the surface syntax for `box <expr>` expressions.
```

Comment:
This comment is now wildly outdated. We haven't had `exchange_malloc` nor `box <expr>` for years.

Reply:
That's from https://github.com/rust-lang/rust/pull/135046 :D

Reply:
It does call `exchange_malloc` though: https://github.com/rust-lang/rust/blob/319f5292a19951d13f3c79c04b3cf3d1501d9e79/compiler/rustc_mir_build/src/builder/expr/as_rvalue.rs?plain=1#L146

Reply:
This intrinsic is how we write `box <expr>`, so in that sense it still exists.

Reply:
`box <expr>` doesn't exist as syntax in the language anymore: https://github.com/rust-lang/rust/pull/108471